Sync ps_get_thread_area() prototype with glibc 2.25.
--- gdb-7.6/gdb/gdb_proc_service.h.orig	2017-04-11 18:57:07.217202451 +0200
+++ gdb-7.6/gdb/gdb_proc_service.h	2017-04-11 19:06:23.527196100 +0200
@@ -115,7 +115,7 @@ extern pid_t ps_getpid (struct ps_procha
 /* Fetch the special per-thread address associated with the given LWP.
    This call is only used on a few platforms (most use a normal register).
    The meaning of the `int' parameter is machine-dependent.  */
-extern ps_err_e ps_get_thread_area (const struct ps_prochandle *,
+extern ps_err_e ps_get_thread_area (struct ps_prochandle *,
 				    lwpid_t, int, psaddr_t *);
 
 
--- gdb-7.6/gdb/i386-linux-nat.c.orig	2017-04-11 18:57:35.960535456 +0200
+++ gdb-7.6/gdb/i386-linux-nat.c	2017-04-11 19:06:54.617195745 +0200
@@ -849,7 +849,7 @@ i386_linux_new_fork (struct lwp_info *pa
    storage (or its descriptor).  */
 
 ps_err_e
-ps_get_thread_area (const struct ps_prochandle *ph, 
+ps_get_thread_area (struct ps_prochandle *ph, 
 		    lwpid_t lwpid, int idx, void **base)
 {
   /* NOTE: cagney/2003-08-26: The definition of this buffer is found
--- gdb-7.6/gdb/aarch64-linux-nat.c.orig	2017-04-11 21:51:39.994282135 +0200
+++ gdb-7.6/gdb/aarch64-linux-nat.c	2017-04-11 22:17:19.040025470 +0200
@@ -750,7 +750,7 @@ aarch64_linux_new_fork (struct lwp_info
    storage (or its descriptor).  */
 
 ps_err_e
-ps_get_thread_area (const struct ps_prochandle *ph,
+ps_get_thread_area (struct ps_prochandle *ph,
 		    lwpid_t lwpid, int idx, void **base)
 {
   struct iovec iovec;
--- gdb-7.6/gdb/amd64-linux-nat.c.orig	2017-04-11 21:51:39.930945697 +0200
+++ gdb-7.6/gdb/amd64-linux-nat.c	2017-04-11 22:16:56.868930201 +0200
@@ -493,7 +493,7 @@ amd64_linux_new_fork (struct lwp_info *p
    a request for a thread's local storage address.  */
 
 ps_err_e
-ps_get_thread_area (const struct ps_prochandle *ph,
+ps_get_thread_area (struct ps_prochandle *ph,
                     lwpid_t lwpid, int idx, void **base)
 {
   if (gdbarch_bfd_arch_info (target_gdbarch ())->bits_per_word == 32)
--- gdb-7.6/gdb/arm-linux-nat.c.orig	2017-04-11 21:51:37.884178700 +0200
+++ gdb-7.6/gdb/arm-linux-nat.c	2017-04-11 22:17:34.610794784 +0200
@@ -613,7 +613,7 @@ supply_fpregset (struct regcache *regcac
 /* Fetch the thread-local storage pointer for libthread_db.  */
 
 ps_err_e
-ps_get_thread_area (const struct ps_prochandle *ph,
+ps_get_thread_area (struct ps_prochandle *ph,
                     lwpid_t lwpid, int idx, void **base)
 {
   if (ptrace (PTRACE_GET_THREAD_AREA, lwpid, NULL, base) != 0)
--- gdb-7.6/gdb/m68klinux-nat.c.orig	2017-04-11 21:51:37.870844712 +0200
+++ gdb-7.6/gdb/m68klinux-nat.c	2017-04-11 22:17:56.778556705 +0200
@@ -568,7 +568,7 @@ fetch_core_registers (struct regcache *r
 /* Fetch the thread-local storage pointer for libthread_db.  */
 
 ps_err_e
-ps_get_thread_area (const struct ps_prochandle *ph, 
+ps_get_thread_area (struct ps_prochandle *ph, 
 		    lwpid_t lwpid, int idx, void **base)
 {
   if (ptrace (PTRACE_GET_THREAD_AREA, lwpid, NULL, base) < 0)
--- gdb-7.6/gdb/mips-linux-nat.c.orig	2017-04-11 21:51:38.040853046 +0200
+++ gdb-7.6/gdb/mips-linux-nat.c	2017-04-11 22:18:09.319176306 +0200
@@ -154,7 +154,7 @@ mips64_linux_register_addr (struct gdbar
 /* Fetch the thread-local storage pointer for libthread_db.  */
 
 ps_err_e
-ps_get_thread_area (const struct ps_prochandle *ph,
+ps_get_thread_area (struct ps_prochandle *ph,
                     lwpid_t lwpid, int idx, void **base)
 {
   if (ptrace (PTRACE_GET_THREAD_AREA, lwpid, NULL, base) != 0)
